hub.py

```diff
@@ -168,49 +183,38 @@ def setup_nfs_share(self):
             f'sudo mkdir -p {self.nfs_share_name} && sudo chown 1000:1000 {self.nfs_share_name} && sudo ls -ld {self.nfs_share_name}'
         ])
 
+
     def deploy(self, auth_provider, proxy_secret_key):
         """
         Deploy this hub
         """
-
-        proxy_secret = hmac.new(proxy_secret_key, self.spec['name'].encode(), hashlib.sha256).hexdigest()
-
-        secret_values = {
-            'jupyterhub':  {
-                'proxy': {
-                    'secretToken': proxy_secret
-                },
+        hub_template = self.spec['template']
+
+        generated_values = self.get_generated_config(auth_provider, proxy_secret_key)
+        # FIXME: Have a templates config somewhere? Maybe in Chart.yaml
+        # FIXME: This is a hack. Fix it.
+        if hub_template != 'base-hub':
+            generated_values = {
+                'base-hub': generated_values
             }
-        }
-
-        # Allow explicilty ignoring auth0 setup
-        if self.spec['auth0'].get('enabled', True):
-            client = auth_provider.ensure_client(
-                self.spec['name'],
-                self.spec['domain'],
-                self.spec['auth0']['connection']
-            )
-            secret_values['jupyterhub']['auth'] = auth_provider.get_client_creds(client, self.spec['auth0']['connection'])
 
+        # FIXME: Don't do this for ephemeral hubs
```

Comment:
Would a template check similar with the one above, be enough?
```python
if hub_template != `ephemeral-hub`:
    self.setup_nfs_share()
```

Reply:
Possibly, but I'd instead like to add an explicit flag to enable / disable NFS. See https://github.com/2i2c-org/pilot-hubs/issues/126